evm/src/NttManager/NttManagerWithPerChainTransceivers.sol

```diff
@@ -0,0 +1,356 @@
+    }
+
+    /// @notice Returns the bitmap of receive transceivers enabled for a chain.
+    /// @param forChainId The chain for which receiving is enabled.
+    function getRecvTransceiverBitmapForChain(
+        uint16 forChainId
+    ) external view returns (uint64) {
+        return _getPerChainTransceiverBitmap(forChainId, RECV_TRANSCEIVER_BITMAP_SLOT);
+    }
+
+    /// @notice Returns the set of chains for which sending is enabled.
+    function getChainsEnabledForSending() external pure returns (uint16[] memory) {
+        return _getEnabledChainsStorage(SEND_ENABLED_CHAINS_SLOT);
+    }
+
+    /// @notice Returns the set of chains for which receiving is enabled.
+    function getChainsEnabledForReceiving() external pure returns (uint16[] memory) {
+        return _getEnabledChainsStorage(RECV_ENABLED_CHAINS_SLOT);
+    }
+
+    /// @notice Returns the number of Transceivers that must attest to a msgId for
+    /// it to be considered valid and acted upon.
+    /// @param forChainId The chain for which the threshold applies.
+    function getThresholdForChain(
+        uint16 forChainId
+    ) public view returns (uint8) {
+        return _getThresholdStoragePerChain()[forChainId].num;
+    }
+
+    // =============== Public Setters ========================================================
+
+    /// @notice Sets the bitmap of transceivers enabled for sending for a chain.
+    /// @param forChainId The chain to be updated.
+    /// @param indexBitmap The bitmap of transceiver indexes that are enabled.
+    function setSendTransceiverBitmapForChain(
+        uint16 forChainId,
+        uint64 indexBitmap
+    ) public onlyOwner {
+        _validateTransceivers(indexBitmap);
+        mapping(uint16 => _EnabledTransceiverBitmap) storage _bitmaps =
+            _getPerChainTransceiverBitmapStorage(SEND_TRANSCEIVER_BITMAP_SLOT);
+        uint64 oldBitmap = _bitmaps[forChainId].bitmap;
+        _bitmaps[forChainId].bitmap = indexBitmap;
+
+        if (indexBitmap == 0) {
+            _removeChain(SEND_ENABLED_CHAINS_SLOT, forChainId);
+        } else {
+            _addChainIfNeeded(SEND_ENABLED_CHAINS_SLOT, forChainId);
+        }
+
+        emit SendTransceiversUpdatedForChain(forChainId, oldBitmap, indexBitmap);
+    }
+
+    /// @notice Sets the bitmap of transceivers enabled for receiving for a chain.
+    /// @param forChainId The chain to be updated.
+    /// @param indexBitmap The bitmap of transceiver indexes that are enabled.
+    /// @param threshold The receive threshold for the chain.
+    function setRecvTransceiverBitmapForChain(
+        uint16 forChainId,
+        uint64 indexBitmap,
+        uint8 threshold
+    ) public onlyOwner {
+        _validateTransceivers(indexBitmap);
+
+        // Validate the threshold against the bitmap.
+        uint8 numEnabled = countSetBits(indexBitmap);
+        if (threshold > numEnabled) {
+            revert ThresholdTooHigh(threshold, numEnabled);
+        }
```

Comment:
Is it worth checking the threshold against the set bits count in the union of the recv bitmap and the global enabled transceiver bitmap? Reduces the probability of setting the wrong bitmap

Reply:
Similarly, if the admin disables a transceiver globally then the per-chain threshold could be > the union of the enabled transceivers. Worth having an equivalent of `_checkTransceiverInvariants` (might need to override again sadly)?